### PR TITLE
Infrastructure

### PR DESCRIPTION
**Pull request by jtomasek**
This change introduces basic infrastucture to use patternfly-react as library in React applications and adds Alert component to test the functionality.

A proper review as well as any feedback is very welcome. Standardjs works wery well except for demanding singlequotes in jsx. That is probably the only rule I'd change. I am still considering if we need to introduce Webpack.

I've tested using Alert component in TripleO-UI and it works very well. To test it with your application
pull down this PR, run `npm install`, `npm run build`. Then in your React application `npm install <path-to-your-patternfly-react-clone>`. Use the Alert component in your app by importing it (`import { Alert } from 'patternfly-react`)

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1 @@
+src
```
Comment (priley86) [inline .npmignore]: I tried to `npm install` , but npm ignoring `src` prevents you from building `src` after `npm install` so I just `cloned` and built instead 😄 .  

Do we want to exclude `src` from our distribution? I suppose `lib` will be built in our release branch?

File: src/Alert/Alert.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import ClassNames from 'classnames'
+import React, { PropTypes } from 'react'
+
+/**
+ * React <b>Alert</b> Component for Patternfly Web Components
```
Comment (priley86) [inline src/Alert/Alert.js]: "<b>Alert</b> Component for Patternfly React"

Do you prefer jsdoc style comments? I have recently been looking at [react-docgen](https://github.com/reactjs/react-docgen). If we are interested, I can give this a try. Seems to maybe be a better tool for pure React component docs.

Referencing: [#1](https://github.com/patternfly/patternfly-react/issues/1)

**Comment by priley86** [comment]
This is a great start @jtomasek. I was able to `import` this Alert component into my app after building `src` with no problem. 

+1 for standardjs. I can work with this.
+1 for Babel. 
+1 for Jest & Jest snapshots

Next thing I would like to think about is how to create test pages so we can demo these PR's to UX & Design in our PRs. Do you have any ideas on this? I think `rawgit` pages with Travis work fine but am open to ideas. I'm also looking to see how we can ensure our website docs stay up to date automatically (with tools like [react-docgen](https://github.com/reactjs/react-docgen)) 

Also hoping to start integrating [semantic-releases](https://github.com/semantic-release/semantic-release) and [coveralls](https://coveralls.io/) for coverage badge. Thoughts?

**Comment by priley86** [comment]
It looks like another potential option for demo'ing our components may be utilizing React [storybooks](https://getstorybook.io/docs/react-storybook/basics/exporting-storybook) which can be exported to gh-pages. Just throwing another idea in the mix to consider...

Any preferences amongst our React devs?

**Comment by priley86** [comment]
I've added React storybooks for this `Alert` component with the commit [here](https://github.com/priley86/patternfly-react/commit/11eb2f5b99d8044bdc6cf5a9a8e4784ed1ddbe40)

You can see it running in my fork's gh-pages here:
https://priley86.github.io/patternfly-react/

Setting this up is actually quite simple. I've included instructions in the README. I think this should be sufficient for demonstrating our new UI components in PRs.

Thoughts on this approach?

File: .npmignore (hunk 1 above)
Comment (jtomasek) [inline .npmignore]: If I am not missing something src should not need to be included in build as it is not needed for consuming app.

File: src/Alert/Alert.js (hunk 2 above)
Comment (jtomasek) [inline src/Alert/Alert.js]: I don't have strict preference. I basically just took what you had in shims repo

**Comment by priley86** [comment]
Updated the storybook to use Knobs and Info add-ons. Have also restructured stories with decorators.

https://github.com/priley86/patternfly-react/commit/0b6fb2a2edbc2a2ee07f1806b68fcc9471898835

Deployed here: https://priley86.github.io/patternfly-react/

any thoughts on this?

**Comment by priley86** [comment]
@jgiardino would love any template ideas ;)

**Comment by jgiardino** [comment]
This is great! I also love using storybook to demo these components.

Some thoughts about the template and documentation shown in storybook:

- I love the knobs tab. As more components get added, it would be great to surface some of the properties on this tab so that someone can see how the component displays as these properties change (e.g. show the Alert with/without the Dismiss button).
  - Related to knobs, if possible it would be great to see this tab display by default, so that someone can easily see what all the options are for a component without having to drill down further.
- I suggest including links to patternfly.org instead of linking to the github page: http://www.patternfly.org/pattern-library/communication/inline-notifications
- I'm in favor of including a link to the design documentation only, and not including any information about what the component does. Since designs could change, including additional information in the storybook potentially adds extra maintenance costs in making sure those contents are still up-to-date.


As for the component, I'm not able to test so I just want to make sure that it allows someone to include a link with the message.

**Comment by jgiardino** [comment]
Oh, meant to say in my previous comment, the css looks as expected.

Thanks for contributing this, @jtomasek!

**Comment by jtomasek** [comment]
Thanks for great feedback! The knobs are certainly useful, although it's usage is a bit limited at the moment as it only supports text, boolean and object type props. Anyway, stories in storybook are meant to actually track all possible props combinations. So IMHO best thing to do is compromise between those two approaches -> use stories in general and knobs for visible changes e.g. to change text in component, make changes in component's style on the fly, toggle the component state (e.g. open/closed dropdown etc.)

I agree with linking the patternfly docs from the storybook and that it is not necessary to duplicate the information there. It should include notes related to ReactJS implementation - in similar manner as 'Code -> Angularjs' tab has in official patternfly docs.

**Comment by priley86** [comment]
@jgiardino it looks like we can make knobs first by just changing the order of `imports` in `addons.js`. I will sync with you about the customizing the template soon @jgiardino :)

**Comment by priley86** [comment]
@jtomasek any chance you can update this Alert component to conditionally support the dismiss icon? This is part of our [feature set](https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/alerts.html) in PF today. Ideally, we can have a fully functional alert when this one gets merged.

I will chat with @jgiardino tomorrow AM about the doc template and try to push some changes. Next thing I will try is integrating semantic releases. My hope is to have a Travis build run on each PR, and once merged, auto release based on semantic release message. Thoughts?

**Comment by jtomasek** [comment]
@priley86 Dismiss icon is optional, if you don't include optional onDismiss prop, the icon will not display.
Agree with all of the next steps, sounds great!

**Comment by jtomasek** [comment]
I've updated this PR to not include snapshots, tests and storybook in build lib. Do you wan't me to update Alert component with react-docgen comments? I am ok with keeping it as part of your pach if you want.

**Comment by priley86** [comment]
@jtomasek good deal. that should reduce npm size 👍 

re: onDismiss prop. great! I will add a storybook story for this in my patch. 

Also, I think it's fine to keep this w/o docgen, and we can merge it with the next PR.

**Comment by priley86** [comment]
repurposing this into #8

**Pull request closed without merging**